File: JavaMiniProjects/BankAccount/src/com/company/Account.java
package com.company;

import java.util.Scanner;

public abstract class Account {
    int balance;

    public void withdrawMoney(int withdrawAmount){
        if (withdrawAmount>this.balance){
            System.out.println("Insufficient fund, The balance in your selected account is " + this.balance);
        }else{
            this.balance -= withdrawAmount;
            System.out.println("You have withdrawn " + withdrawAmount + " and the balance in your selected account is " + this.balance);
        }
    }

    public void depositMoney(int depositAmount){
        if(depositAmount<0){
            System.out.println("Amount cannot be negative");
        }else{
            this.balance += depositAmount;
            System.out.println("Amount deposited successfully. The balance in your selected account is " + this.balance);
        }
    }

    public void checkBalance(){
    }

    public void run(){
        boolean innerCond = true;
        while(innerCond){
            Scanner sc = new Scanner(System.in);
            System.out.println("Select action from below");
            if (this.balance == 0){
                System.out.println("\n1.deposit\n2.checkBalance\n3.Main menu");
                int choice = sc.nextInt();
                if(choice==1){
                    System.out.println("Specify the amount to deposit");
                    int depositAmount = sc.nextInt();
                    depositMoney(depositAmount);
                }else if(choice == 2){
                    checkBalance();
                }else if (choice == 3){
                    innerCond = false;
                }
            }else{
                System.out.println("\n1.withdraw\n2.deposit\n3.checkBalance\n4.Main menu");
                int choice = sc.nextInt();
                if(choice == 1){
                    System.out.println("Specify the amount to withdraw");
                    int withdrawAmount = sc.nextInt();
                    withdrawMoney(withdrawAmount);
                }else if(choice == 2){
                    System.out.println("Specify the amount to deposit");
                    int depositAmount = sc.nextInt();
                    depositMoney(depositAmount);
                }else if (choice == 3){
                    checkBalance();
                }else if (choice == 4){
                    innerCond = false;
                }
            }
        }
    }
}

File: python_projects/bankMiniProject/sample/account.py
from abc import ABC, abstractmethod #abstract module used for declaring abstract class...

class Account(ABC):
    balance = 0
    depositChoice = ""
    
    @classmethod
    @abstractmethod
    def checkBalance(cls):
        pass
    
    @classmethod
    def amountToWithdraw(cls):
        print("\nEnter the Amount to Withdraw.\n")
        withdrawAmount = int(input())
        if(withdrawAmount > cls.balance or cls.balance == 0):
            print("Insifficient Fund!!")
            print("\nWould you like to deposit Money?(y or n)")
            cls.depositChoice = str(input())
            if(cls.depositChoice == 'y'):
                cls.amountToDeposit()
        else:
            cls.balance -= withdrawAmount
            print("Amount withdrawn successfully!!")

    @classmethod
    def amountToDeposit(cls):
        while(cls.depositChoice == 'y'):
            print("\nEnter the Amount to Deposit.\n")
            depositAmount = int(input())
            if(depositAmount > 0):
                cls.balance += depositAmount
                print("Amount deposited successfully!!")
                cls.depositChoice = 'n'
            else:
                print("Amount deposit unsuccessful!!.")
                print("Would you like to try again (y or n)")
                cls.depositChoice = str(input())
    @classmethod
    def mainMethod(cls):
        while(True):
            print("\nPlease choose from option below(press 1 or 2 or 3):\n1.Withdraw Money\n2.Deposit Money\n3.Check Balance\n4.Return to Main Menu")
            actionSelected = int(input())
            if(actionSelected == 1):
                cls.amountToWithdraw()
            elif(actionSelected == 2):
                cls.depositChoice = "y"
                cls.amountToDeposit()
            elif(actionSelected == 3):
                cls.checkBalance()
            elif(actionSelected == 4):
                break
            else:
                print("\nTry Again\n.")
    

